packages/test-base/src/androidAndroidTest/kotlin/io/realm/kotlin/test/shared/CopyFromRealmTests.kt

```diff
@@ -0,0 +1,755 @@
+        }
+
+        val managedSample1 = realm.writeBlocking {
+            copyToRealm(sample)
+        }
+        val managedSample2 = realm.writeBlocking {
+            findLatest(managedSample1)!!.apply {
+                intField = 42
+            }
+        }
+        val o1p = (managedSample1 as RealmObjectInternal).io_realm_kotlin_objectReference!!.objectPointer
+        val o2p = (managedSample2 as RealmObjectInternal).io_realm_kotlin_objectReference!!.objectPointer
+        assertFalse(RealmInterop.realm_equals(o1p, o2p))
+
+        val unmanagedObjects = realm.copyFromRealm(listOf(managedSample1, managedSample2))
+        assertEquals(2, unmanagedObjects.size)
+        val unmanagedSample1 = unmanagedObjects.first()
+        val unmanagedSample2 = unmanagedObjects.last()
+
+        assertNotSame(unmanagedSample1, unmanagedSample2)
+        assertNotEquals(unmanagedSample1, unmanagedSample2)
+        assertEquals("v1", unmanagedSample1.stringField)
+        assertEquals(3, unmanagedSample1.intField)
+        assertEquals(42, unmanagedSample2.intField)
+    }
+
+    @Test
+    fun depth_nullAfterDepthIsReached() {
+        val sample = Sample().apply {
+            stringField = "obj-depth-0"
+            nullableObject = Sample().apply {
+                stringField = "obj-depth-1"
+                nullableObject = Sample().apply {
+                    stringField = "obj-depth-2"
+                }
+            }
+            objectListField = realmListOf(
+                Sample().apply {
+                    stringField = "list-depth-1"
+                    objectListField = realmListOf(
+                        Sample().apply {
+                            stringField = "list-depth-2"
+                        }
+                    )
+                }
+            )
+            objectSetField = realmSetOf(
+                Sample().apply {
+                    stringField = "set-depth-1"
+                    objectSetField = realmSetOf(
+                        Sample().apply {
+                            stringField = "set-depth-2"
+                        }
+                    )
+                }
+            )
+        }
+
+        val managedObj = realm.writeBlocking {
+            copyToRealm(sample)
+        }
+        assertEquals("obj-depth-2", managedObj.nullableObject!!.nullableObject!!.stringField)
+        assertEquals("list-depth-2", managedObj.objectListField.first().objectListField.first().stringField)
+        assertEquals("set-depth-2", managedObj.objectSetField.first().objectSetField.first().stringField)
+
+        val unmanagedCopy = managedObj.copyFromRealm(depth = 1u)
+        assertEquals("obj-depth-1", unmanagedCopy.nullableObject!!.stringField)
+        assertEquals("list-depth-1", unmanagedCopy.objectListField.first().stringField)
+        assertEquals("set-depth-1", unmanagedCopy.objectSetField.first().stringField)
+        assertNull(unmanagedCopy.nullableObject!!.nullableObject)
```

Comment:
I guess this would also be my standard expectation, but seeing it against the copied string fields makes me wonder ... 🤔 ... but until we could express an _unset_ property or something like that it is maybe better just to `null` it 😅